02.calendar/cal.rb

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'date'
+require 'optparse'
+
+class Calendar
+  DAY_OF_WEEK = 'Su Mo Tu We Th Fr Sa'
+  PRINT_WIDTH = DAY_OF_WEEK.length
+  TODAY = Date.today
+
+  attr_reader :target_date, :target_year, :target_month
+
+  def initialize(options:)
+    @target_year = options[:y]&.to_i || TODAY.year
+    @target_month = options[:m]&.to_i || TODAY.month
+    @target_date = Date.new(@target_year, @target_month, 1)
+  end
+
+  def print_calendar
+    print_headers
+    print_numbers
+  end
+
+  private
+
+  def print_headers
+    puts target_date.strftime('%B %Y').center(PRINT_WIDTH)
+    puts DAY_OF_WEEK
+  end
+
+  def print_numbers
+    first_day = Date.new(target_year, target_month, 1)
+    last_day = Date.new(target_year, target_month, -1)
+
+    print '   ' * first_day.wday
+    first_day.step(last_day) do |date|
+      print formatted_day(date:)
+      puts if date.saturday?
+    end
+    puts
+  end
+
+  def formatted_day(date:)
```

Comment:
```suggestion
  def formatted_day(date)
```

でいいですね

あと、メソッド名は動詞で始めましょう

https://qiita.com/jnchito/items/459d58ba652bf4763820#%E5%87%A6%E7%90%86%E3%82%92%E5%AE%9F%E8%A1%8C%E3%81%99%E3%82%8B%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89%E3%81%AF-%E5%8B%95%E8%A9%9E%E3%81%AE%E3%81%BF-or-%E5%8B%95%E8%A9%9E--%E5%90%8D%E8%A9%9E%E3%81%AB%E3%81%99%E3%82%8B

Reply:
素直に`format_day`という名前に変更しました。

ちなみに、どこかで読んだ「インスタンス変数っぽく定義する」というのをやってみたつもりなのですが、このケースだと不適切だったでしょうか？（メソッド内の処理が多い or インプットとアウトプットの差が大きいなど？）

Reply:
> 「インスタンス変数っぽく定義する」というのをやってみたつもり

たしかにそういうテクニックはあります。
その場合は引数がないというのが原則です。

```ruby
# これはインスタンス変数（プロパティ）っぽい
some_date.formatted_day

# これはインスタンス変数（プロパティ）っぽくない
some_date.formatted_day("%s")
```

引数があるとインスタンス変数っぽくない（そもそも変数に引数なんて渡せない）ので、そういう場合は動詞で始めてちゃんとメソッドらしく命名した方がいいです。

```ruby
some_date.format_day("%s")
```

Reply:
> 引数があるとインスタンス変数っぽくない（そもそも変数に引数なんて渡せない）ので、

ものすごく納得しました...！フィードバックありがとうございます🙇‍♀️